pipelines/api/utils.py

```diff
@@ -77,12 +79,25 @@ def _slugify_file(filename):
     basename = filename.rsplit('/', 1)[-1]
     return basename.rsplit('.', 1)[0]
 
-
 def _run_id_iterator(slug):
     for sub_folder in os.listdir(slug):
         if _is_valid_uuid(sub_folder):
             yield sub_folder
 
+# patched with limit, order by modify time
+def _run_id_iterator_with_limit(slug, limit):
+    prefix = slug.rstrip('/')
+    tgt = glob.glob('{}/*'.format(prefix))
+    # filter valid uuid
+    it = filter(lambda x: _is_valid_uuid(os.path.basename(x)), tgt)
```

Comment:
need to make sure the item in `tgt` is a dir as well

Reply:
make sure `tgt` are directories too.